testutil/beaconmock/options.go

```diff
@@ -242,6 +242,58 @@ func WithSlotsPerEpoch(slotsPerEpoch int) Option {
 	}
 }
 
+// WithDutiesPubkeys configures the mock to return attestation, sync committee and proposer
+// duties with the prpovided public key.
+func WithDutiesPubkeys(key eth2p0.BLSPubKey) Option {
```

Comment:
adding this to beaconmock for one specific test feels a bit much, rather only do this in the test itself.